pkg/istio-agent/agent.go

```diff
@@ -228,9 +240,91 @@ func (a *Agent) initializeEnvoyAgent() error {
 
 	drainDuration, _ := types.DurationFromProto(a.proxyConfig.TerminationDrainDuration)
 	a.envoyAgent = envoy.NewAgent(envoyProxy, drainDuration)
+	a.envoyWaitCh = make(chan error, 1)
+	if a.cfg.EnableDynamicBootstrap {
+		// Simulate an xDS request for a bootstrap
+		go func() {
+			// wait indefinitely and keep retrying with jittered exponential backoff
+			backoff := 500
+			max := 30000
+			request := &bootstrapDiscoveryRequest{
+				node:        node,
+				envoyWaitCh: a.envoyWaitCh,
+				envoyUpdate: envoyProxy.UpdateConfig,
+			}
+			for {
```

Comment:
during this loop agent is ignoring SIGTERM/SIGINT which may block shut down, potentially indefinitely. Can we add some early cancellation?